app/javascript/components/KeywordSearch.js

```diff
@@ -1,62 +1,39 @@
-import React, { Component } from 'react';
+import React from 'react';
 import Button from 'react-bootstrap/lib/Button';
 import InputGroup from 'react-bootstrap/lib/InputGroup';
 import Form from 'react-bootstrap/lib/Form';
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 
 
-class KeyWordSearch extends React.Component{
-  constructor(props){
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this)
-    this.state = {
-      searchTerms: undefined,
-    };
-  };
+export default function KeywordSearch (props){
 
-  handleSubmit(e){
+  const handleSubmit = (event) => {
     // Prevent full page reload
-    e.preventDefault();
-
-    const searchTerm = e.target.elements.searchText.value.trim();
+    event.preventDefault();
+    const searchTerm = event.target.elements.keywordText.value.trim();
```

Comment:
Let's get the value from the form element directly with document.getElementById('keyword-input')